main.tf

```diff
@@ -1,9 +1,118 @@
+data "archive_file" "this" {
+  type        = "zip"
+  source_file = "${path.module}/cloudflareupdater.py"
+  output_path = "${path.module}/tmp/cloudflareupdater.zip"
+}
+
+data "aws_iam_policy_document" "assume" {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "this" {
+  name_prefix        = var.name
+  assume_role_policy = data.aws_iam_policy_document.assume.json
```

Comment:
a before n

Reply:
not when it's name, name or name equivalent can always go first